app/c2/gist.py

```diff
@@ -0,0 +1,151 @@
+
+    def __init__(self, services, module, config, name):
+        self.key = config.get('key')
+        super().__init__(services, module, config, name)
+
+    def start(self, app):
+        loop = asyncio.get_event_loop()
+        loop.run_forever(self._default_active_c2_loop())
+
+    def get_config(self):
+        """
+        Returns this C2 objects api key
+        :return: GIST api key
+        """
+        return 'githubToken', self.key
+
+    async def get_results(self):
+        """
+        Retrieve all GIST posted results for a this C2's api key
+        :return:
+        """
+        results = await self._get_raw_gist_urls(comm_type='results')
+        result_content = await self._get_gist_content([result[0] for result in results])
+        await self._delete_gists([result[1] for result in results])
+        return result_content
+
+    async def get_beacons(self):
+        """
+        Retrieve all GIST beacons for a particular api key
+        :return: the beacons
+        """
+        beacons = await self._get_raw_gist_urls(comm_type='beacon')
+        beacon_content = await self._get_gist_content([beacon[0] for beacon in beacons])
+        await self._delete_gists([beacon[1] for beacon in beacons])
+        return beacon_content
+
+    async def post_payloads(self, payloads, paw):
+        """
+        Given a list of payloads and an agent paw, posts the payloads as a series of base64 encoded
+        files attached to a single gist
+        :param payloads:
+        :param paw:
+        :return:
+        """
+
+        files = {payload[0]: dict(content=self._encode_string(payload[1])) for payload in payloads}
+        if len(files) < 1 or await self._wait_for_paw(paw, comm_type='payloads'):
+            return
+        gist = self._build_gist_content(comm_type='payloads', paw=paw, files=files)
+        return await self._post_gist(gist)
+
+    async def post_instructions(self, text, paw):
+        """
+        Post an instruction for the agent to execute as an encoded GIST
+        :param text: The instruction text for the agent to execute
+        :param paw: The paw for the agent to execute
+        :return:
+        """
+        if len(json.loads(self.decode_bytes(text))['instructions']) < 1 or \
+                await self._wait_for_paw(paw, comm_type='instructions'):
+            return
+        gist = self._build_gist_content(comm_type='instructions', paw=paw, files={str(uuid.uuid4()): {"content": text}}
+                                        )
+        return await self._post_gist(gist)
+
+    """ PRIVATE """
+
+    async def _post_gist(self, gist):
+        headers = {'Authorization': 'token {}'.format(self.key)}
+        async with aiohttp.ClientSession(headers=headers, connector=aiohttp.TCPConnector(verify_ssl=False)) as session:
+            return await self._post(session, 'https://api.github.com/gists', body=gist)
+
+    @staticmethod
+    def _build_gist_content(comm_type, paw, files):
+        return {
```

Comment:
use single quotes. on all dicts in this class, use dict() format instead of {}

Reply:
done for where possible.  Not possible in all cases:
`files = {payload[0]: dict(content=self._encode_string(payload[1])) for payload in payloads}`
and
`{str(uuid.uuid4()): {'content': text}}` for example